query-coordinator/src/main/scala/com/socrata/querycoordinator/resources/QueryResource.scala

```diff
@@ -341,7 +347,7 @@ class QueryResource(secondary: Secondary,
             val (finalSchema, analyses) = versionInfo.payload
             val (rewrittenAnalyses, rollupName) = possiblyRewriteOneAnalysisInQuery(finalSchema, analyses)
```

Comment:
At some point this will have to know about context variables or about whatever we want to call functions that can return different output based on things other than the data.  If the rollup code knew a function was of the latter category it could easily refuse to rewrite anything using such a function.

Reply:
Ah true, I'll have to consider that.  I'd like to have a general notion of "impure" function to support things like timestamps too...

Reply:
It is easy to have the rollup code just hardcode this one function for now but having a general class of functions sounds good and would make it easier to have all the places that may care not need updating with new ones in the future.